src/test/java/net/snowflake/client/jdbc/ResultSetStructuredTypesLatestIT.java

```diff
@@ -204,6 +216,21 @@ public void testReturnAsArrayOfSqlData() throws SQLException {
         });
   }
 
+  @Test
+  public void testReturnAsArrayOfNullableFieldsInSqlData() throws SQLException {
+    SnowflakeObjectTypeFactories.register(SimpleClass.class, FewTypesSqlData::new);
+    withFirstRow(
+        "SELECT OBJECT_CONSTRUCT_KEEP_NULL('string', null, 'nullableIntValue', null, 'nullableLongValue', null, 'longValue', null)::OBJECT(string VARCHAR, nullableIntValue INTEGER, nullableLongValue INTEGER, longValue INTEGER)",
+        (resultSet) -> {
+          FewTypesSqlData result =
+              resultSet.unwrap(SnowflakeBaseResultSet.class).getObject(1, FewTypesSqlData.class);
+          assertNull(result.getString());
+          assertNull(result.getNullableIntValue());
+          assertNull(result.getNullableLongValue());
+          assertEquals(Long.valueOf(0), result.getLongValue());
```

Comment:
Why we have 0 here?

Reply:
It depends of reading implemented in SQLData. I've set this field as a primitive